src/assets/styles/styles.11ty.js

```diff
@@ -0,0 +1,192 @@
+                    } else {
+                        const processedCss = this.processCss({str: css});
+                        resolve(processedCss);
+                    }
+                });
+        });
+    }
+
+    // Process CSS with PostCSS.
+    async processCss(options) {
+        const css = postcss([autoprefixer]).process(options.str, {
+            from: options.entryPath || false,
+            map: {
+                inline: true
+            }
+        }).css;
+        return css;
+    }
+
+    // Compile and process CSS (source-agnostic).
+    async compile(options) {
+        let css;
+        switch (options.fileType) {
+        case "css":
+            css = await this.processCss(options);
+            break;
+        case "scss":
+            css = await this.compileScss(options);
+            break;
+        case "styl":
+            css = await this.compileStylus(options);
+            break;
+        }
+        return css;
+    }
+
+    // Minify & optimize with CleanCSS in production.
+    async minify(css) {
+        return new Promise((resolve, reject) => {
+            if (!isProd) {
+                resolve(css);
+            }
+
+            const minified = new CleanCSS().minify(css);
+            if (minified.styles) {
+                resolve(minified.styles);
+            } else {
+                reject(minified.error);
+            }
+        });
+    }
+
+    // Display an error overlay when CSS build fails.
+    // this brilliant idea is taken from Mike Riethmuller / Supermaya
+    // @see https://github.com/MadeByMike/supermaya/blob/master/site/utils/compile-scss.js
+    renderError(error) {
+        return `
+        /* Error compiling stylesheet */
+        *,
+        *::before,
+        *::after {
+            box-sizing: border-box;
+        }
+        html,
+        body {
+            margin: 0;
+            padding: 0;
+            min-height: 100vh;
+            font-family: monospace;
+            font-size: 1.25rem;
+            line-height:1.5;
+        }
+        body::before {
+            content: "";
+            background: #000;
+            top: 0;
+            bottom: 0;
+            width: 100%;
+            height: 100%;
+            opacity: 0.7;
+            position: fixed;
+        }
+        body::after {
+            content: "${cssesc(error)}";
+            white-space: pre;
+            display: block;
+            top: 0;
+            padding: 30px;
+            margin: 50px;
+            width: calc(100% - 100px);
+            color:#721c24;
+            background: #f8d7da;
+            border: solid 2px red;
+            position: fixed;
+        }`;
+    }
+
+    // Render the CSS file.
+    async render({entryPath}) {
+        try {
+            const fileType = entryPath.split(".").pop();
+            const str = fs.readFileSync(entryPath, "utf8");
+            const css = await this.compile({fileType, entryPath, str});
+            return await this.minify(css);
```

Comment:
I think you can just return `this.minify(css)` without using `await`. If I'm not mistaken it will return the promise from `this.minify` if you use `await`, I suppose it will return a new promise which will resolve with the result of `this.minify`.

Reply:
Resolved in 3bbeccd.